implementation/src/main/java/io/smallrye/metrics/legacyapi/LegacyMetricRegistryAdapter.java

```diff
@@ -84,6 +86,10 @@ public MeterRegistry getPrometheusMeterRegistry() {
         return registry;
     }
 
+    /**
+     * Calls to this method should check {@link #isAppNameResolverPresent}
+     * to avoid overhead of having to convert {@link MetricDescriptor} to {@link MetricID}
+     */
```

Comment:
comment is a bit unclear on intent.  maybe say "This method should not be called when {@link #isAppNameResolverPresent} is false as this method does nothing in that case."